### PR TITLE
Add New NEOS Setup

### PR DESCRIPTION
**Pull request by valish**

### CODE REVIEW

File: js/store/environment/actions.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
+import moment from 'moment';
```
Comment (andyh2) [inline js/store/environment/actions.js]: I think we can remove the entire `environment` store, not using date or location if I'm reading correctly

Comment (andyh2) [reply js/store/environment/actions.js]: And also perhaps remove moment.js dependency. It's big

File: js/store/index.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    debug: true,
+});
+
+const log = (state = {}, action) => {
+    console.log('dispatched', action.type);
+    return state
+};
+
+export default combineReducers({
+    environment: persistReducer(persistConfig('environment'), environment),
```
Comment (andyh2) [inline js/store/index.js]: Remove redux-persist

File: js/store/params/actions.js [hunk 3]
```diff
@@ -0,0 +1,59 @@
+import ChffrPlus from '../../native/ChffrPlus';
```
Comment (andyh2) [inline js/store/params/actions.js]: Remove entire params store

File: js/timers/UpdateChecker.js [hunk 4]
```diff
@@ -0,0 +1,25 @@
+import ChffrPlus from '../native/ChffrPlus';
```
Comment (andyh2) [inline js/timers/UpdateChecker.js]: Remove this file

File: js/timers/UploadProgressTimer.js [hunk 5]
```diff
@@ -0,0 +1,31 @@
+import { ACTION_UPLOAD_SUMMARY_UPDATED } from '../store/uploads/actions';
```
Comment (andyh2) [inline js/timers/UploadProgressTimer.js]: Remove this file

File: js/utils/AppStateListener.js [hunk 6]
```diff
@@ -0,0 +1,26 @@
+import { AppState } from 'react-native';
```
Comment (andyh2) [inline js/utils/AppStateListener.js]: Remove this file

File: js/utils/bytes.js [hunk 7]
```diff
@@ -0,0 +1,18 @@
+export const formatSize = (bytes) => {
```
Comment (andyh2) [inline js/utils/bytes.js]: Remove this file

File: js/utils/conversions.js [hunk 8]
```diff
@@ -0,0 +1,5 @@
+export const farenToCel = (f) => (f - 32) * (5/9);
```
Comment (andyh2) [inline js/utils/conversions.js]: Remove this file

File: js/utils/number.js [hunk 9]
```diff
@@ -0,0 +1,3 @@
+export const round = (number, places) => {
```
Comment (andyh2) [inline js/utils/number.js]: Remove this, seems unused

File: js/utils/pluralize.js [hunk 10]
```diff
@@ -0,0 +1,7 @@
+export default function(str, count) {
```
Comment (andyh2) [inline js/utils/pluralize.js]: Remove this, is unused

File: js/utils/url.js [hunk 11]
```diff
@@ -0,0 +1,8 @@
+export function queryStringFromObj(obj) {
```
Comment (andyh2) [inline js/utils/url.js]: Remove

File: js/utils/version.js [hunk 12]
```diff
@@ -0,0 +1,53 @@
...
+
+let _isPassive = false;
+let initPromise = init();
+
+async function init() {
+  const paramIsPassive = await ChffrPlus.readParam(Params.KEY_IS_PASSIVE);
+  _isPassive = paramIsPassive === "1";
+}
+
+export const checkHasCompletedTraining = async () => {
```
Comment (andyh2) [inline js/utils/version.js]: I think this whole file can be removed.

File: android/app/src/main/java/ai/comma/plus/neossetup/ChffrPlusModule.kt [hunk 13]
```diff
@@ -0,0 +1,440 @@
+package ai.comma.plus.neossetup
```
Comment (andyh2) [inline android/app/src/main/java/ai/comma/plus/neossetup/ChffrPlusModule.kt]: Lots of unused code here. Strip down this file to only the methods used.

File: build.sh [hunk 14]
```diff
@@ -0,0 +1,49 @@
...
+cd $(dirname $0)
+
+APK_OUT=${1:-ai.comma.plus.neossetup.apk}
+SIGNAPK=${SIGNAPK:="./signapk"}
+CEREAL=${CEREAL:="./cereal"}
+export SENTRY_SKIP_UPLOAD=${SENTRY_SKIP_UPLOAD:-1}
+
+if [ ! -d $SIGNAPK ]; then
+  echo "sign apk not found"
+  git clone https://github.com/techexpertize/SignApk.git $SIGNAPK
```
[the remark this answers is not in the document]
Comment (valish) [reply build.sh]: @quillford

**Pull request merged**